langchain/llms/azureml_endpoint.py

```diff
@@ -0,0 +1,236 @@
+        """Initialize the class."""
+        self.endpoint_url = endpoint_url
+        self.endpoint_api_key = endpoint_api_key
+        self.deployment_name = deployment_name
+    
+    def call(self, body):
+        """call."""
+
+        url = self.endpoint_url
+        # Replace this with the primary/secondary key or AMLToken for the endpoint
+        api_key = self.endpoint_api_key
+        if not api_key:
+            raise Exception("A key should be provided to invoke the endpoint")
+
+        # The azureml-model-deployment header will force the request to go to a specific deployment.
+        # Remove this header to have the request observe the endpoint traffic rules
+
+        headers = {'Content-Type': 'application/json', 'Authorization': ('Bearer ' + api_key), 'azureml-model-deployment': self.deployment_name}
+        
+        req = urllib.request.Request(url, body, headers)
+        try:
+            response = urllib.request.urlopen(req, timeout=50)
+            result = response.read()
+        except urllib.error.HTTPError as error:
+            print("The request failed with status code: " + str(error.code))
+            result = str(error)
+        except Exception as e:
+            print("Calling Azure Managed Online endpoint failed!")
+            result = str(e)
+        return result
+
+class ContentFormatterBase():
+    """A handler class to transform input from LLM to
+    a format that AzureML endpoint expects.
+    """
+
+    """
+    Example:
+        .. code-block:: python
+
+            class ContentFormatter(ContentFormatterBase):
+                content_type = "application/json"
+                accepts = "application/json"
+                
+                def format_request_payload(self, prompt, model_kwargs) -> bytes:
+                    input_str = json.dumps({"inputs": {"input_string": [prompt]}, "parameters": model_kwargs})
+                    return str.encode(input_str)
+
+                def format_response_payload(self, output) -> str:
+                    response_json = json.loads(output)
+                    return response_json[0]["0"]
+    """
+    content_type: Optional[str] = "text/plain"
+    """The MIME type of the input data passed to the endpoint"""
+    
+    accepts: Optional[str] = "text/plain"
+    """The MIME type of the response data returned form the endpoint"""
+
+    @abstractmethod
+    def format_request_payload(self, prompt, model_kwargs) -> bytes:
+        """Formats the request body according to the input schema of 
+        the model. Returns bytes or seekable file like object in the
+        format specified in the content_type request header.
+        """
+
+    @abstractmethod
+    def format_response_payload(self, output) -> Any:
+        """Formats the response body according to the output
+        schema of the model. Returns the data type that is
+        received from the response.
+        """
+     
+class LLMContentFormatter(ContentFormatterBase):
```

Comment:
What's the motivation for another later between the specific formatters and the ContentFormatterBase?

Reply:
I don't think it is necessary. I was trying something similar to the sagemaker endpoint, but does not do anything special. I will remove it and use `ContentFormatterBase` as the superclass of the other content formatters rather than `LLMContentFormatter`